src/v/pandaproxy/schema_registry/seq_writer.h

```diff
@@ -0,0 +1,88 @@
+//// Copyright 2021 Vectorized, Inc.
+////
+//// Use of this software is governed by the Business Source License
+//// included in the file licenses/BSL.md
+////
+//// As of the Change Date specified in that file, in accordance with
+//// the Business Source License, use of this software will be governed
+//// by the Apache License, Version 2.
+
+#pragma once
+
+#include "kafka/client/client.h"
+#include "pandaproxy/schema_registry/sharded_store.h"
+#include "pandaproxy/schema_registry/types.h"
+#include "random/simple_time_jitter.h"
+#include "utils/retry.h"
+
+namespace pandaproxy::schema_registry {
+
+using namespace std::chrono_literals;
+
+class seq_writer {
+public:
+    seq_writer(ss::sharded<kafka::client::client>& client, sharded_store& store)
+      : _client(client)
+      , _store(store) {}
+
+    ss::future<> read_sync();
+
+    ss::future<schema_id>
+    write_subject_version(subject sub, schema_definition def, schema_type type);
+
+private:
+    ss::future<> read_sync_inner();
+
+    ss::sharded<kafka::client::client>& _client;
+    sharded_store& _store;
+
+    struct WriteCollision : public std::exception {
+        const char* what() const throw() { return "Write Collision"; }
+    };
+
+    simple_time_jitter<ss::lowres_clock> _jitter{
+      std::chrono::milliseconds{100}};
+
+    /// Helper for write paths that use sequence+retry logic to synchronize
+    /// multiple writing nodes.
+    template<typename F>
+    ss::future<
+      typename std::invoke_result_t<F, model::offset>::value_type::value_type>
+    sequenced_write(F f) {
+        auto _units = co_await _store.begin_write();
+        auto base_backoff = _jitter.next_duration();
+
+        co_return co_await retry_with_backoff(
+          std::numeric_limits<int>::max(),
```

Comment:
I didn't see this. There's probably a reasonable retries, and timeout in the Kafka/client config, but it could also be explicit config in schema_registry for conflict scenerios.

Reply:
I was going for "no timeout" when writing this initially.  It makes sense to add a retry limit to catch bugs though, so I'll do that.